### PR TITLE
Don't reload data after onDestroyView

### PR DESCRIPTION
**Pull request by vycius**
Fixes https://console.firebase.google.com/u/1/project/tvarkau-vilniu/monitoring/app/android:lt.vilnius.tvarkau/cluster/2b9f8b8f?duration=2592000000
https://console.firebase.google.com/u/0/project/tvarkau-vilniu/monitoring/app/android:lt.vilnius.tvarkau/cluster/53385b7e?pli=1&reportTypes=REPORT_TYPE_UNSPECIFIED&duration=2592000000

### CODE REVIEW

File: app/src/main/java/lt/vilnius/tvarkau/fragments/BaseReportListFragment.kt [hunk 1]
```diff
@@ -52,16 +53,26 @@ abstract class BaseReportListFragment : BaseFragment(), ReportListView {
                 .subscribeOn(ioScheduler)
                 .observeOn(uiScheduler)
                 .subscribe({
-                    reloadData()
+                    reloadReports = true
```
Comment (mjurkus) [inline app/src/main/java/lt/vilnius/tvarkau/fragments/BaseReportListFragment.kt]: While the fix would work, I think it's a hack :)

`onError` handler is needed here.

And unsubscribe should be before `super.onDestroyView()`. That should be enough I think.

Comment (vycius) [reply app/src/main/java/lt/vilnius/tvarkau/fragments/BaseReportListFragment.kt]: This won't work. If unsubscribtion is done in `onDestroyView` ReportsList fragment won't receive an event of applied filters from ReportsFilterFragment.  
We need to keep subscribed `RxBus` after `onDestroyView`.

Comment (mjurkus) [reply app/src/main/java/lt/vilnius/tvarkau/fragments/BaseReportListFragment.kt]: True.

Still `onError` is needed here.

Comment (vycius) [reply app/src/main/java/lt/vilnius/tvarkau/fragments/BaseReportListFragment.kt]: True. I will add that.

**Comment by vycius** [comment]
@mjurkus Added on error handler

**Review by mjurkus — APPROVED** [review]

**Pull request merged**